cloud-platform/env-create.md

```diff
@@ -0,0 +1,232 @@
+
+#### Buildspec
+
+The `buildspec.yaml` file contains codebuild specification that will create resources defined in the namespaces.
+
+#### Namespace
+
+The namespaces contains subdirectories named after each of the desired namespaces you want to create.
+
+You will be working within the namespace directory to create your environment.
+
+### GitHub directory structure
+
+#### k8s-non-prod-environment
+
+![Image](images/image4.png)
+
+This is the root of the repo, containing Terraform and Namespace directory
+
+#### /namespaces
+
+![Image](images/image5.png)
+
+This is the Namespace directory, this is where you will add create a directory for your service in the format `$servicename-$env`   
+example: myapp-dev
+
+#### /$Servicename-$env
+
+![Image](images/image6.png)
+
+When you create your `$servicename-$env` directory for your service. You will need to create two files within it. `Namespaces.yaml` and `$servicename-$env-admin-role.yaml`
+
+### Instructions
+
+![Image](images/image2.png)
+
+1) Git clone the repo onto your local machine and create a new branch to make your changes. (Using terminal)
+
+```
+  #git clone the repo onto your local machine.
+  $ git clone git@github.com:ministryofjustice/k8s-nonprod-environments.git
+
+  # change directory into the k8s-non prod repo
+  cd k8s-nonprod-environments/
+
+  #create new branch in the repo, my-app is used as an example, you can call it something descriptive.
+  $ git checkout -b my-app
+
+  #you can confirm you are using the branch you made using the command.
+  $ git branch -a
+
+  #Current branch you are currently using will be indicated by the *.
+
+    master
+  * my-app
+    remotes/origin/HEAD -> origin/master
+    remotes/origin/add-git-crypt
+```
+
+2) Now we need to amend the namespaces directory in the root of repo and create an directory within it.
+
+```
+#we will need to change directories and change into the namespaces directory.
+cd namespaces/
+
+#we shall now be in our namespaces directory but we can confirm by the following command.
+pwd
+
+#output should show that we are in the namespaces directory. This will all be dependent on where you have chosen to git clone the repo. In this example k8s-nonprod was cloned into the Git directory which is located in Documents.
+
+$ pwd
+/Users/<username>/Desktop/Git/k8s-nonprod-environments/namespaces
+
+#Now we must create the directory for our service, the name of the directory we create should be in the format $servicename-$env
+$ mkdir myapp-dev
+
+#Now that we have created our namespaces subdirectory $servicename-env, lets now change into this directory.
+$ cd myapp-dev
+```
+
+3) Now we will need to create two files within our `$servicename-$env` directory.  `Namespaces.yaml` and `$servicename-$env-admin-role.yaml`. The structures of both files are shown below.
+
+
+#### k8s-non-prod-environment/namespaces/$servicename-$env/namespace.yaml
+
+```
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: myapp-dev
+  labels:
```

Comment:
I don't think we need to have users create a `name` label, as there's already a `name` attribute on the namespace, unless we're doing something specific with the label elsewhere?

We might want to consider adding labels for other reasons though, much like the AWS tagging policy:
```
metadata:
  labels:
    org: laa
    service: fee-calculator
    environment: staging
```

That might be too much too soon though?